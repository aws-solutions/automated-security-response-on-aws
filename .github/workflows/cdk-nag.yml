name: CDK Nag Test

on:
  push:
  pull_request:
    types: [opened, edited, reopened, synchronize]

jobs:
  cdk-nag:
    name: CDK Nag Check
    if: ${{ (github.repository_owner != 'aws-solutions' && github.event_name == 'push') || (github.repository_owner == 'aws-solutions' && github.event_name == 'pull_request') }}
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [16.x]
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
      - run: |
          cd source
          npm ci
          cd solution_deploy
          npx cdk synth
          cd ../playbooks/AFSBP
          npx cdk synth
          cd ../CIS120
          npx cdk synth
          cd ../CIS140
          npx cdk synth
          cd ../NEWPLAYBOOK
          npx cdk synth
          cd ../PCI321
          npx cdk synth
          cd ../SC
          npx cdk synth
