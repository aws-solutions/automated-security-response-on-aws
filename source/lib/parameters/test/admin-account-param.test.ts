// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { App, Aspects, DefaultStackSynthesizer, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { AwsSolutionsChecks } from 'cdk-nag';
import AdminAccountParam from '../admin-account-param';

function createAdminAccountParamStack(): Stack {
  const app = new App();
  const stack = new Stack(app, 'AdminAccountParamStack', {
    analyticsReporting: false,
    synthesizer: new DefaultStackSynthesizer({ generateBootstrapVersionRule: false }),
  });
  Aspects.of(stack).add(new AwsSolutionsChecks({ verbose: true }));
  new AdminAccountParam(stack, 'AdminAccountParam');
  return stack;
}

describe('admin account param stack', function () {
  const template = Template.fromStack(createAdminAccountParamStack());

  it('matches snapshot', function () {
    expect(template).toMatchSnapshot();
  });

  describe('template param', function () {
    const allowedPattern = /^\d{12}$/;

    it('is present', function () {
      template.hasParameter('SecHubAdminAccount', {
        AllowedPattern: allowedPattern.source,
        Type: 'String',
      });
    });

    describe('allowed pattern', function () {
      it('matches account number', function () {
        expect('1'.repeat(12)).toMatch(allowedPattern);
      });

      it('does not match too long account number', function () {
        expect('1'.repeat(13)).not.toMatch(allowedPattern);
      });

      it('does not match words', function () {
        expect('MyAdminAccnt').not.toMatch(allowedPattern);
      });
    });
  });
});
