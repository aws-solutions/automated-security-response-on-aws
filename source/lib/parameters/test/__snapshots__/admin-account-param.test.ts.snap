// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin account param stack matches snapshot 1`] = `
{
  "Parameters": {
    "SecHubAdminAccount": {
      "AllowedPattern": "^\\d{12}$",
      "Description": "Admin account number",
      "Type": "String",
    },
  },
}
`;
